### PR TITLE
Remove unused pulpcore.plugin objects

### LINKED ISSUE #5693: pulp-workers should log which domain a task was executed in
When a task id is referenced in the pulp-worker logs, that ID is not enough information to look up the task status from the REST API. The task id needs to be accompanied by a domain name also.

### PR DESCRIPTION
**Pull request by bmbouter**
This was done by an audit against master for pulp_rpm, pulp_container,
pulp_file, pulp_deb, pulp-certguard, and pulp_cookbook.

pulp_container will require a small update replacing its CharInFilter
usage.

https://pulp.plan.io/issues/5693
closes #5693

### CODE REVIEW

**Comment by bmbouter** [comment]
pulp_container needs an update replacing its https://github.com/pulp/pulp_container/blob/c924075a7d6817840cf6a58d324680170dd09564/pulp_container/app/viewsets.py#L42 with direct usage of https://github.com/pulp/pulpcore/blob/4029a85f5950ad4b33bf83dc23f5c7c8fff97c9b/pulpcore/app/viewsets/custom_filters.py#L249

**Comment by goosemania** [comment]
@bmbouter , 
RPM plugin uses `MultipleArtifactContentSerializer` and `Model`. 
Migration plugin uses `Model`, `IdentityField`, `HyperlinkRelatedFilter`.

`Model` will become `BaseModel`, and what to do with the rest? Direct import?

Let's coordinate the merge/cherry-pick, so RPM plugin is not broken for long.

File: CHANGES/plugin_api/5693.removal [hunk 1]
```diff
@@ -0,0 +1 @@
+The ``pulpcore.plugin.models.Model`` is renamed to ``pulpcore.plugin.models.BaseModel``.
```
Comment (goosemania) [inline CHANGES/plugin_api/5693.removal]: I think it's better to mention all the objects which are removed from the plugin API, so plugin writers are not surprised that something disappeared.

**Comment by ipanova** [comment]
@bmbouter CharInFilter is used by pulp_container.

**Comment by ipanova** [comment]
@bmbouter pulp_container uses CharInFilter, since

> pulp_container needs an update replacing its https://github.com/pulp/pulp_container/blob/c924075a7d6817840cf6a58d324680170dd09564/pulp_container/app/viewsets.py#L42 with direct usage of
> 
> https://github.com/pulp/pulpcore/blob/4029a85f5950ad4b33bf83dc23f5c7c8fff97c9b/pulpcore/app/viewsets/custom_filters.py#L249

ok

File: CHANGES/plugin_api/5693.removal (hunk 1 above)
Comment (bmbouter) [reply CHANGES/plugin_api/5693.removal]: Good idea! I will add those here.

**Comment by bmbouter** [comment]
> @bmbouter ,
> RPM plugin uses `MultipleArtifactContentSerializer` and `Model`.
> Migration plugin uses `Model`, `IdentityField`, `HyperlinkRelatedFilter`.

Let me back out these changes

> 
> `Model` will become `BaseModel`, and what to do with the rest? Direct import?

Yes I'm going to try to open PRs to fix this before merging this.

> 
> Let's coordinate the merge/cherry-pick, so RPM plugin is not broken for long.

:+1:

**Comment by bmbouter** [comment]
I added back `MultipleArtifactContentSerializer`, `IdentityField`, `HyperlinkRelatedFilter`, and `CharInField` to keep things simple. So really the main change is the `Model` -> `BaseModel`. That is motivated because our API shouldn't compete with Django for the actual `Model` (django is the real one).

**Comment by bmbouter** [comment]
I also added the removals in release notes.

**Comment by bmbouter** [comment]
Here's the corresponding pulp_rpm PR: https://github.com/pulp/pulp_rpm/pull/1546 I'm making one for pulp_container now too.

**Comment by bmbouter** [comment]
I realized pulp_container didn't need an update, but pulp-2to3-migraiton does: https://github.com/pulp/pulp-2to3-migration/pull/53

**Pull request merged**